### PR TITLE
Log request params for errored requests.

### PR DESCRIPTION
**Pull request by andrewplummer**

### CODE REVIEW

File: src/loggers/GoogleCloudLogger.js [hunk 1]
```diff
@@ -54,7 +54,9 @@ export default class GoogleCloudLogger extends BaseLogger {
     this.emitPayload({
       message,
       severity,
-      ...this.getAdditionalFields(info),
+      userId: info.userId,
```
Comment (kaareal) [inline src/loggers/GoogleCloudLogger.js:57]: how do you set the userId

File: src/middleware.js [hunk 2]
```diff
@@ -65,3 +66,44 @@ function getRequestInfo(ctx) {
...
+  if (ctx.status >= 400) {
+    const { body, query } = ctx.request;
+    return {
+      requestBody: strip(body, options),
+      requestQuery: strip(query, options),
+    };
+  } else {
+    return {};
+  }
+}
```
Comment (kaareal) [inline src/middleware.js]: nice, i would make it perhaps only 500?

**Review by kaareal — APPROVED** [review]

File: src/loggers/GoogleCloudLogger.js (hunk 1 above)
Comment (andrewplummer) [reply src/loggers/GoogleCloudLogger.js:57]: This already existed before (just moved around). It's captured from the `authUser` on the request

File: src/middleware.js (hunk 2 above)
Comment (andrewplummer) [reply src/middleware.js]: Part of the point of this is to be able to get insights into any request failures so 400 level is just as important, maybe more so if we're doing probably error handling, which we tend to

**Comment by andrewplummer** [comment]
OK so I updated this to only allow recording of params (body or query for GET) by explicit inclusion:

```js
app.use(logger.middleware({
   recordParams: [
    {
      // Record all params for this route with status >= 400
      path: '/1/foo',  
    },
    {
      // Record all params for this route with status >= 200
      status: 200,  
      path: '/1/foo',  
    },
    {
      // Record only POST requests
      method: 'POST',
      path: '/1/foo',  
    },
    {
      // Match path by regex
      path: /\/1\//foo/,  
    },
    {
      // Include only specific params
      include: ['shop']
    },
    {
      // Include all but excluded params
      exclude: ['shop']
    },
    {
      // Params also support regexes
      include: [/shop/]
    },
  ],
}));
```

```js
// Records all request >= (over status 400 by default)
app.use(logger.middleware({
   recordParams: true,
}));
```

**Review by dominiek — APPROVED** [review]

**Pull request merged**